File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    
    let map = new Map();
    
    for(let num of array) {
        let v = map.has(num) ? map.get(num) : 0;
        map.set(num, v+1);
    }
    
    var answer = -1;
    var max = 0;
    var over = false;
    
    for(let [key, value] of map) {
        if(value > max) {
            max = value;
            answer = key;
            over = false;
        }
        else if(value == max) {
            over = true;
        }
    }
    
    return over ? -1 : answer;
}

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.java
import java.util.HashMap;
import java.util.Map.Entry;

class Solution {
    public int solution(int[] array) {
        
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for(int num : array) {
            map.put(num, map.getOrDefault(num,0)+1);
        }
        
        int answer = -1;
        int max = 0;
        boolean over = false;
        
        for(Entry<Integer,Integer> entry : map.entrySet()) {
            if(entry.getValue() > max) {
                answer = entry.getKey();
                max = entry.getValue();
                over = false;
            }
            else if(entry.getValue() == max) {
                over = true;
            }
        }
        
        return over ? -1 : answer;
    }
}